### PR TITLE
Add Euclidean GCD C++ Code

### PR DESCRIPTION
**Pull request by RohanGoyal2014**
**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

Fixes #{ISSUE_NUMBER}

**By submitting this pull request I confirm I've read and complied with the
below declarations.**

- [x] I have read the [Contribution guidelines](https://github.com/NITSkmOS/Algorithms/blob/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [x] I have followed the [commit guidelines](https://github.com/NITSkmOS/Algorithms/blob/master/CONTRIBUTING.md#write-good-commit-messages) for this project.
- [x] My code follows the [standard code structure](https://github.com/NITSkmOS/Algorithms/blob/master/CONTRIBUTING.md#samples-code).
- [x] This pull request has a descriptive title. For example, `{Tag}: Add {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.
- [x] This pull request will be closed if I fail to update it even once in a continuous time span of 7 days.
- [x] This pull request shall only be reviewed and merged once all the checks passes. No maintainer or supporter shall be obliged to review it before this condition is met.
- [x] I have mentioned the issue number correctly (with hyperlink) in this pull request description.

After you submit your pull request, **DO NOT** click the 'Update Branch' button.

### CODE REVIEW

**Comment by sangamcse** [comment]
@sangamcse rebase

**Comment by sangamcse** [comment]
Hey! I'm [GitMate.io](https://gitmate.io)! This pull request is being rebased automatically. Please **DO NOT** push while rebase is in progress or your changes would be lost permanently :warning:

**Comment by sangamcse** [comment]
Automated rebase with [GitMate.io](https://gitmate.io) was successful! :tada:

**Comment by sangamcse** [comment]
Comment on 7f2a83cff83d2900951f05bfa33f6dff6531afdf, file euclidean_gcd/C++/euclid.cpp, line 47.

Missing spaces around =  [whitespace/operators] [4]

*Origin: CPPLintBear, Section: `all.cpplint`.*

**Comment by sangamcse** [comment]
Comment on 7f2a83cff83d2900951f05bfa33f6dff6531afdf, file euclidean_gcd/C++/euclid.cpp, line 62.

An else should appear on the same line as the preceding }  [whitespace/newline] [4]

*Origin: CPPLintBear, Section: `all.cpplint`.*

**Comment by sangamcse** [comment]
Comment on 7f2a83cff83d2900951f05bfa33f6dff6531afdf, file euclidean_gcd/C++/euclid.cpp, line 62.

If an else has a brace on one side, it should have it on both  [readability/braces] [5]

*Origin: CPPLintBear, Section: `all.cpplint`.*

**Comment by sangamcse** [comment]
Comment on 7f2a83cff83d2900951f05bfa33f6dff6531afdf, file euclidean_gcd/C++/euclid.cpp, line 63.

Missing spaces around <<  [whitespace/operators] [3]

*Origin: CPPLintBear, Section: `all.cpplint`.*

**Comment by sangamcse** [comment]
Comment on 9205fec90cf8b625c442df02efc6db156f9d4cff, file euclidean_gcd/C++/euclid.cpp, line 62.

An else should appear on the same line as the preceding }  [whitespace/newline] [4]

*Origin: CPPLintBear, Section: `all.cpplint`.*

**Comment by sangamcse** [comment]
Comment on 9205fec90cf8b625c442df02efc6db156f9d4cff, file euclidean_gcd/C++/euclid.cpp, line 62.

If an else has a brace on one side, it should have it on both  [readability/braces] [5]

*Origin: CPPLintBear, Section: `all.cpplint`.*

**Comment by sangamcse** [comment]
Comment on 9205fec90cf8b625c442df02efc6db156f9d4cff, file euclidean_gcd/C++/euclid.cpp, line 63.

Missing spaces around <<  [whitespace/operators] [3]

*Origin: CPPLintBear, Section: `all.cpplint`.*

File: euclidean_gcd/C++/euclid.cpp [hunk 1]
```diff
@@ -0,0 +1,65 @@
+#include<bits/stdc++.h>
+using namespace std;
+
+int euclidean_gcd_division_based(int a, int b) {
+    /*
+     * Calculates gcd of two numbers using iterative euclid division method
+     */
```
Comment (sangamcse) [inline euclidean_gcd/C++/euclid.cpp]: Missing spaces around <<  [whitespace/operators] [3]

*Origin: CPPLintBear, Section: `all.cpplint`.*

**Comment by RohanGoyal2014** [comment]
@sangamcse  please look into it.

**Pull request closed without merging**